BitFaster.Caching/Lfu/CmSketchCore.cs

```diff
@@ -231,40 +235,28 @@ private void Reset()
         }
 
 #if !NETSTANDARD2_0
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        //[MethodImpl((MethodImplOptions)512)]
         private unsafe int EstimateFrequencyAvx(T value)
         {
             int blockHash = Spread(comparer.GetHashCode(value));
             int counterHash = Rehash(blockHash);
             int block = (blockHash & blockMask) << 3;
 
-            Vector128<int> h = Vector128.Create(counterHash);
-            h = Avx2.ShiftRightLogicalVariable(h.AsUInt32(), Vector128.Create(0U, 8U, 16U, 24U)).AsInt32();
+            Vector128<int> h = Avx2.ShiftRightLogicalVariable(Vector128.Create(counterHash).AsUInt32(), Vector128.Create(0U, 8U, 16U, 24U)).AsInt32();
+            Vector128<int> index = Avx2.ShiftLeftLogical(Avx2.And(Avx2.ShiftRightLogical(h, 1), Vector128.Create(15)), 2);
+            Vector128<int> blockOffset = Avx2.Add(Avx2.Add(Vector128.Create(block), Avx2.And(h, Vector128.Create(1))), Vector128.Create(0, 2, 4, 6));
 
-            var index = Avx2.ShiftRightLogical(h, 1);
-            index = Avx2.And(index, Vector128.Create(15)); // j - counter index
-            Vector128<int> offset = Avx2.And(h, Vector128.Create(1));
-            Vector128<int> blockOffset = Avx2.Add(Vector128.Create(block), offset); // i - table index
-            blockOffset = Avx2.Add(blockOffset, Vector128.Create(0, 2, 4, 6)); // + (i << 1)
+            Vector256<ulong> indexLong = Avx2.PermuteVar8x32(Vector256.Create(index, Vector128<int>.Zero), Vector256.Create(0, 4, 1, 5, 2, 5, 3, 7)).AsUInt64();
```

Comment:
Can this become readonlyspan<byte> as suggested here for the permute lookup table:

https://bits.houmus.org/2020-01-30/this-goes-to-eleven-pt3